### PR TITLE
fixed checked-call test bug associated to newer versions of numpy

### LINKED ISSUE #792: Running py.test fails epically
On my Ubuntu 14.04 system running the 3.15 linux kernel, running all the tests using py.test fails horribly.

```
sean@hybrid:~/GitHub-linux/nengo_spa_fix$ py.test --pyargs nengo
============================= test session starts ==============================
platform linux2 -- Python 2.7.6 -- py-1.4.30 -- pytest-2.7.2
rootdir: /home/sean/GitHub-linux/nengo_spa_fix, inifile: setup.cfg
collected 630 items 

nengo/networks/tests/test_actionselection.py ..
nengo/networks/tests/test_circularconv.py .......
nengo/networks/tests/test_ensemblearray.py .........
nengo/networks/tests/test_integrator.py .
nengo/networks/tests/test_oscillator.py .
nengo/networks/tests/test_product.py .
nengo/networks/tests/test_workingmemory.py .
nengo/spa/tests/test_action_objects.py ........
nengo/spa/tests/test_actions.py ......
nengo/spa/tests/test_assoc_mem.py .....
nengo/spa/tests/test_basalganglia.py ..
nengo/spa/tests/test_buffer.py ....
nengo/spa/tests/test_compare.py ..
nengo/spa/tests/test_cortical.py ......
nengo/spa/tests/test_input.py ...
nengo/spa/tests/test_memory.py .....
nengo/spa/tests/test_pointer.py .................
nengo/spa/tests/test_spa.py .
nengo/spa/tests/test_thalamus.py s...
nengo/spa/tests/test_vocabulary.py .........
nengo/tests/test_base.py ...
nengo/tests/test_builder.py .........
nengo/tests/test_cache.py ........................s
nengo/tests/test_config.py .......
nengo/tests/test_connection.py ..................................................................................
nengo/tests/test_dists.py ......................
nengo/tests/test_ensemble.py ..........................................................
nengo/tests/test_examples.py ..........................ssssssssssssssssFF........................................
nengo/tests/test_learning_rules.py ...................
nengo/tests/test_network.py .....
nengo/tests/test_neurons.py ..................
nengo/tests/test_node.py ..............
nengo/tests/test_params.py ............
nengo/tests/test_probe.py .........
nengo/tests/test_processes.py ..............
nengo/tests/test_pytest.py .
nengo/tests/test_simulator.py ......
nengo/tests/test_solvers.py ..........s.......ssssssssssss
nengo/tests/test_synapses.py .........
nengo/utils/tests/test_builder.py .
nengo/utils/tests/test_builder_graphviz.py .
nengo/utils/tests/test_builder_passthrough.py ...
nengo/utils/tests/test_cache.py ...
nengo/utils/tests/test_compat.py .
nengo/utils/tests/test_connection.py ....................
nengo/utils/tests/test_ensemble.py .................
nengo/utils/tests/test_filter_design.py ....
nengo/utils/tests/test_functions_piecewise.py .......
nengo/utils/tests/test_graphs.py ...
nengo/utils/tests/test_logging.py ..
nengo/utils/tests/test_magic.py ......
nengo/utils/tests/test_matplotlib.py ss
nengo/utils/tests/test_nco.py .......
nengo/utils/tests/test_network.py .
nengo/utils/tests/test_neurons.py ..s
nengo/utils/tests/test_numpy.py .
nengo/utils/tests/test_probe.py ....
nengo/utils/tests/test_progress.py ...........
nengo/utils/tests/test_stdlib.py F.....
nengo/utils/tests/test_testing.py ......

=================================== FAILURES ===================================
_ test_nooutput[/home/sean/GitHub-linux/nengo_spa_fix/examples/integrator-2d] __

nb_file = '/home/sean/GitHub-linux/nengo_spa_fix/examples/integrator-2d'

    @pytest.mark.example
    @pytest.mark.parametrize('nb_file', all_examples)
    def test_nooutput(nb_file):
        """Ensure that no cells have output."""
        pytest.importorskip("IPython", minversion="1.0")
        pytest.importorskip("jinja2")
        from nengo.utils.ipython import load_notebook
    
        def check_all(cells):
            for cell in cells:
                if cell.cell_type == 'code':
                    assert cell.outputs == [], ("Clear outputs in %s" % nb_path)
    
        nb_path = os.path.join(examples_dir, "%s.ipynb" % nb_file)
        nb = load_notebook(nb_path)
        if nb.nbformat <= 3:
            for ws in nb.worksheets:
>               check_all(ws.cells)

nengo/tests/test_examples.py:100: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cells = [{u'source': u'# Nengo Example: Integrator\n\nThis demo implements a one-dimens...ations (e.g. soft normalization).', ...tadata': {}}, {u'source': u'## Step 3: Connect the network elements', u'cell_type': u'markdown', u'metadata': {}}, ...]

    def check_all(cells):
        for cell in cells:
            if cell.cell_type == 'code':
>               assert cell.outputs == [], ("Clear outputs in %s" % nb_path)
E               AssertionError: Clear outputs in /home/sean/GitHub-linux/nengo_spa_fix/examples/integrator-2d.ipynb
E               assert [{u'output_ty...da7dc02d90>'}] == []
E                 Left contains more items, first extra item: {u'output_type': u'display_data', u'javascript': u'\n        require(["widgets...metadata': {}, u'text': u'<IPython.core.display.Javascript at 0x7fda7dc02d90>'}
E                 Use -v to get the full diff

nengo/tests/test_examples.py:94: AssertionError
_ test_nooutput[/home/sean/GitHub-linux/nengo_spa_fix/examples/integrator-EA] __

nb_file = '/home/sean/GitHub-linux/nengo_spa_fix/examples/integrator-EA'

    @pytest.mark.example
    @pytest.mark.parametrize('nb_file', all_examples)
    def test_nooutput(nb_file):
        """Ensure that no cells have output."""
        pytest.importorskip("IPython", minversion="1.0")
        pytest.importorskip("jinja2")
        from nengo.utils.ipython import load_notebook
    
        def check_all(cells):
            for cell in cells:
                if cell.cell_type == 'code':
                    assert cell.outputs == [], ("Clear outputs in %s" % nb_path)
    
        nb_path = os.path.join(examples_dir, "%s.ipynb" % nb_file)
        nb = load_notebook(nb_path)
        if nb.nbformat <= 3:
            for ws in nb.worksheets:
>               check_all(ws.cells)

nengo/tests/test_examples.py:100: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cells = [{u'source': u'# Nengo Example: Integrator\n\nThis demo implements a one-dimens...ations (e.g. soft normalization).', ...tadata': {}}, {u'source': u'## Step 3: Connect the network elements', u'cell_type': u'markdown', u'metadata': {}}, ...]

    def check_all(cells):
        for cell in cells:
            if cell.cell_type == 'code':
>               assert cell.outputs == [], ("Clear outputs in %s" % nb_path)
E               AssertionError: Clear outputs in /home/sean/GitHub-linux/nengo_spa_fix/examples/integrator-EA.ipynb
E               assert [{u'output_ty...869bbc9e10>'}] == []
E                 Left contains more items, first extra item: {u'output_type': u'pyout', u'prompt_number': 24, u'metadata': {}, u'text': u'<matplotlib.legend.Legend at 0x7f869bb5fc10>'}
E                 Use -v to get the full diff

nengo/tests/test_examples.py:94: AssertionError
______________________________ test_checked_call _______________________________

    def test_checked_call():
        def func1(a):
            return a
    
        def func2(a, b=0, **kwargs):
            return a+b
    
        def func3(a, b=0, c=0, *args, **kwargs):
            return a+b+c+sum(args)
    
        func4 = lambda x=[0]: sum(x)
    
        class A(object):
            def __call__(self, a, b):
                return a + b
    
        assert checked_call(func1) == (None, False)
        assert checked_call(func1, 1) == (1, True)
        assert checked_call(func1, 1, 2) == (None, False)
        assert checked_call(func1, 1, two=2) == (None, False)
    
        assert checked_call(func2) == (None, False)
        assert checked_call(func2, 1) == (1, True)
        assert checked_call(func2, 1, 2) == (3, True)
        assert checked_call(func2, 1, 2, three=3) == (3, True)
        assert checked_call(func2, 1, 2, 3) == (None, False)
    
        assert checked_call(func3) == (None, False)
        assert checked_call(func3, 1) == (1, True)
        assert checked_call(func3, 1, 2) == (3, True)
        assert checked_call(func3, 1, 2, 3) == (6, True)
        assert checked_call(func3, 1, 2, 3, 4, 5, 6) == (21, True)
    
        assert checked_call(func4) == (0, True)
        assert checked_call(func4, [1, 2]) == (3, True)
        assert checked_call(func4, [1], 2) == (None, False)
    
        assert checked_call(A(), 1) == (None, False)
        assert checked_call(A(), 1, 2) == (3, True)
        assert checked_call(A(), 1, 2, 3) == (None, False)
    
        assert checked_call(np.sin) == (None, False)
        assert checked_call(np.sin, 0) == (0, True)
>       assert checked_call(np.sin, 0, np.array([1])) == (np.array([0]), True)
E       assert CheckedCall(v...invoked=False) == (array([0]), True)
E         At index 0 diff: None != array([0])
E         Use -v to get the full diff

nengo/utils/tests/test_stdlib.py:50: AssertionError
============== 3 failed, 593 passed, 34 skipped in 148.92 seconds ==============
```

These seem to be pretty simple stdlib functions. Any idea why they might be failing?

### PR DESCRIPTION
**Pull request by Seanny123**
Fixes #792 .

### CODE REVIEW

File: nengo/utils/tests/test_stdlib.py [hunk 1]
```diff
@@ -47,8 +47,8 @@ def __call__(self, a, b):
 
     assert checked_call(np.sin) == (None, False)
     assert checked_call(np.sin, 0) == (0, True)
-    assert checked_call(np.sin, 0, np.array([1])) == (np.array([0]), True)
-    assert checked_call(np.sin, 0, np.array([1]), 1) == (None, False)
+    assert checked_call(np.sin, 0, np.array([1.])) == (np.array([0]), True)
```
Comment (jgosmann) [inline nengo/utils/tests/test_stdlib.py]: The RHS should probably say `np.array([0.])` too, shouldn't it?

Comment (Seanny123) [inline nengo/utils/tests/test_stdlib.py]: For some reason that didn't throw an error, but to be consistent I'll change it in the next commit.

Comment (jgosmann) [inline nengo/utils/tests/test_stdlib.py]: Yeah, I suppose for comparison the cast is still done automatically. But consistency is good. :)

**Comment by jgosmann** [comment]
ping @Seanny123 Are going to fix the commit message? I think, otherwise than that this is good to be merged.

**Comment by Seanny123** [comment]
ah, dammit, I totally forgot, I'll fix it after the meeting today

**Comment by Seanny123** [comment]
@jgosmann fixed the commit messsage

**Comment by tbekolay** [comment]
Looks good, merging!

**Pull request closed without merging**